src/Simulation/Common/QubitManager.cs

```diff
@@ -355,91 +494,90 @@ public virtual void Release(IQArray<Qubit> qubitsToRelease)
 
             for (long i = qubitsToRelease.Length-1; i>=0; i--)
             { // Going from the end is more efficient in case we are tracking scope.
-                this.ReleaseOneQubit(qubitsToRelease[i], usedOnlyForBorrowing: false);
+                this.Release(qubitsToRelease[i], wasUsedOnlyForBorrowing: false);
             }
         }
 
-        protected virtual void DisableOneQubit(Qubit qubit)
-        {
-            // Note: Borrowed qubits cannot be disabled.
-            Debug.Assert(qubits[qubit.Id] == Allocated);
-            qubits[qubit.Id] = Disabled;
-            numDisabledQubits++;
+        #endregion
 
-            numAllocatedQubits--;
-            Debug.Assert(numAllocatedQubits >= 0);
-        }
+        #region Borrow
 
         /// <summary>
-        /// Disables a given qubit.
-        /// Once a qubit is disabled it can never be reallocated.
+        /// Returns a qubit that is free, not disabled, and not listed as in use, whose id is in [minId, maxId).
```

Comment:
It looks like it actually will never return a free qubit, only an allocated qubit that's available for borrowing (not in scope in the current frame).

Reply:
fixed